.github/workflows/asdf_ci.yml

```diff
@@ -7,41 +7,141 @@ on:
     branches: [ master ]
 
 jobs:
-  lint:
-    runs-on: ubuntu-latest
-
-    steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.8
-    - name: Install Dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install flake8
-    - name: Lint with flake8
-      run: |
-        flake8 asdf --count --show-source --statistics
-
-  build:
-    runs-on: ubuntu-latest
+  tox:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        python-version: ['3.8']
+        include:
+          - name: Python 3.9 Testing
+            os: ubuntu-latest
+            python-version: 3.9
+            toxenv: py39
+
+          - name: Python 3.8 Testing
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: py38
+
+          - name: Python 3.7 Testing
+            os: ubuntu-latest
+            python-version: 3.7.9
+            toxenv: py37
+
+          - name: Python 3.6 Testing
+            os: ubuntu-latest
+            python-version: 3.6.10
+            toxenv: py36
+
+          - name: Documentation Build
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: docbuild
+
+          - name: Big-endian platform
+            os: ubuntu-latest
+            python-version: 3.9
+            with: 
+              arch: s390x
+            toxenv: s390x
+
+          - name: Windows
+            os: windows-latest
+            python-version: 3.9
+            toxenv: py39
+
+          - name: Mac OS Latest
+            os: macos-latest
+            python-version: 3.9
+            toxenv: py39
+
+          - name: Compatibility 
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: compatibility
+
+          - name: Bandit Security Checks
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: bandit
+
+          - name: Code Coverage
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: coverage
```

Comment:
This is fine as a replacement for the current Travis CI matrix, but I think coverage should be folded into the `py39` test above.  And also perhaps the `py36` test above.  Running the exact same tests here just for coverage is redundant.

Reply:
I removed the py39 test and changed the code coverage test to 3.9.